### DISCUSSION TITLE
Clarification on maxUniqueItems in Amazon SP-API JSON Schema (MOTORCYCLE_PART)

### DISCUSSION THREAD

**Discussion by rdzs2** [amzn/selling-partner-api-models#4363] [General]
Hello,

I am working on dynamically generating a user interface based on the JSON schema provided by the Amazon SP-API, specifically for the product type MOTORCYCLE_PART.

I have a question regarding the `maxUniqueItems `property defined for some attributes:

1. For product_description, the schema specifies `maxUniqueItems`: 1000. Does this mean I could theoretically provide up to 1000 text boxes for describing the product?
2. For bullet_point, the value of `maxUniqueItems` is 10, which seems reasonable.
3. However, for language, the `maxUniqueItems` value is set to 103, which seems excessive for a practical use case.

Despite these values, in Amazon Seller Central, I notice that these fields (e.g., `product_description, language`) only allow adding up to 5 text boxes.

Am I misunderstanding the purpose of `maxUniqueItems`, or is there an additional constraint not reflected in the schema? Could it be a limitation imposed by the UI in Seller Central that differs from the schema?

I would appreciate any clarification on this discrepancy.

Thank you!

**Comment by weilinggu** [comment]
Hi @rdzs2,
As defined in [Amazon Product Type Definitions Meta-Schema](https://developer-docs.amazon.com/sp-api/docs/product-type-definition-meta-schema), maxUniqueItems indicates the maximum number of items with the same combination of selector properties so your understanding is correct, you can provide up to X values for the attribute you are looking at.

If you are uncertain whether if there is additional downstream constraint not reflected in the schema, you can use [validation preview feature](https://developer-docs.amazon.com/sp-api/docs/sp-api-release-notes#listings-items-api-v2021-08-01-1). This feature allows you to preview listing errors in production without creating listings.

The experience in UI via Seller Central and in schema via the API will continue to get consolidated in the future so they provide a consistent experience.

Thanks
Weiling
Amazon Selling Partner Services

**Reply by rdzs2** [reply]
Thank you!
